python/pyarrow/ipc.py

```diff
@@ -119,3 +120,40 @@ def open_file(source, footer_offset=None):
     reader : RecordBatchFileReader
     """
     return RecordBatchFileReader(source, footer_offset=footer_offset)
+
+
+def serialize_pandas(df):
+    """Serialize a pandas DataFrame into a buffer protocol compatible object.
+
+    Parameters
+    ----------
+    df : pandas.DataFrame
+
+    Returns
+    -------
+    buf : buffer
+        An object compatible with the buffer protocol
+    """
+    batch = pa.RecordBatch.from_pandas(df)
+    sink = pa.InMemoryOutputStream()
+    writer = pa.RecordBatchFileWriter(sink, batch.schema)
+    writer.write_batch(batch)
+    writer.close()
+    return sink.get_result()
+
+
+def deserialize_pandas(buf):
```

Comment:
can you add `nthreads=None` here and pass through to `to_pandas` (single-threaded by default)